### ISSUE TITLE
Using Hash#dig only works in Ruby 2.3.x or later

### ISSUE THREAD

**Issue by sfisher** [datacite/omniauth-orcid#11]
We ended up having to upgrade Ruby to use this gem because the method #dig is not present until the Ruby 2.3 series and later.  We were running Ruby 2.2.x.  Dig forced an upgrade to Ruby (which then forced an upgrade to Rails because of an Integer bug in the version of Rails on a newer Ruby version).

The documentation even shows the use of Ruby 1.9.3 (where Hash#dig doesn't work).

```[2012-11-26 21:41:08] INFO  ruby 1.9.3 (2012-04-20) [x86_64-darwin11.3.0]
== Sinatra/1.3.2 has taken the stage on 4567 for development with backup from WEBrick
[2012-11-26 21:41:08] INFO  WEBrick::HTTPServer#start: pid=8383 port=4567
```

You may want to rethink using this new Hash core library feature if you want better compatibility with other versions of Ruby or else declare Ruby 2.3+ as a dependency for people looking to use the gem.

**Comment by mfenner** [comment]
Thank you. Looking at https://www.ruby-lang.org/en/downloads/branches/, it seems reasonable to require Ruby 2.2, but not 2.3. It is easy to revert that change. And I need to update the documentation (which is from 2012).

**Comment by sfisher** [comment]
Thanks, Martin.

**Comment by nicolasfranck** [comment]
I added the gem ruby_dig to make this method work.

**Comment by nicolasfranck** [comment]
https://github.com/datacite/omniauth-orcid/pull/13

**Comment by mfenner** [comment]
Closed via #13.
